Runtime.php

```diff
@@ -79,20 +79,23 @@ public static function getUri()
      */
     public static function getData($extended = true)
     {
-        switch (static::getMethod()) {
+        $method = static::getMethod();
+
+        switch ($method) {
             case Request::METHOD_GET:
                 return $_GET;
 
             case Request::METHOD_POST:
+            case Request::METHOD_PUT:
+            case Request::METHOD_PATCH:
                 $contentType = static::getHeader('Content-Type');
 
                 switch ($contentType) {
                     case 'application/x-www-form-urlencoded':
-                        return $_POST;
+                        return ${'_'.strtoupper($method)};
 
                     case 'application/json':
                         $input = file_get_contents('php://input');
-
                         if (true !== $extended ||
                             true !== function_exists('json_decode')) {
```

Comment:
So the behavior seems not correct: if the user passes `$extended = true` he expects that the JSON gets parsed. If there is no `json_decode` in this case I would rather throw an exception as the client code might get an unexpected string.